examples/3_write_serial.cpp

```diff
@@ -63,15 +66,17 @@ int main(int argc, char *argv[])
     rho.resetDataset(dataset);
     cout << "Set the dataset properties for the scalar field rho in iteration 1\n";
 
-    series.flush();
+    // in FlushType::DEFER, call now:
+    // series.flush();
     cout << "File structure and required attributes have been written\n";
 
     Offset offset = {0, 0};
     rho.storeChunk(shareRaw(global_data), offset, extent);
     cout << "Stored the whole Dataset contents as a single chunk, "
             "ready to write content\n";
 
-    series.flush();
+    // in FlushType::DEFER, call now:
+    // series.flush();
```

Comment:
This commented out `flush()` is very likely what's causing the SEGFAULT.
In `RecordComponent::store_chunk()`, you only schedule a single `DATASET_WRITE` (well, multiple ones potentially).
Literally no other IO operations have been scheduled or completed at that point. No HDF5 handle exists, no structure inside the HDF5 file. So naturally the call in the backend fails ([here](https://github.com/openPMD/openPMD-api/blob/0.7.1-alpha/src/IO/HDF5/HDF5IOHandler.cpp#L670-L673)).

Reply:
> This commented out `flush()` is very likely what's causing the SEGFAULT.

Yes, I know, that's why I was wondering if we want to [re-structure some of the logic](https://github.com/openPMD/openPMD-api/pull/484#issuecomment-466646641) in side `Record::flush_impl` so `RecordComponent` knows about it?